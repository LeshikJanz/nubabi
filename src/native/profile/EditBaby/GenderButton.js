// @flow
import type { Gender } from '../../../common/types';
import React from 'react';
import { View, StyleSheet, Text, TouchableOpacity } from 'react-native';

type Props = {
  buttonText: string,
  selected: boolean,
  onChangeGender: (gender: Gender) => void,
};

const GenderButton = ({ buttonText, selected, onChangeGender }: Props) => {
  return (
    <TouchableOpacity onPress={onChangeGender}>
      <View
        style={[
          styles.genderButton,
          { marginRight: 20 },
          selected ? styles.selectedButton : '',
        ]}
      >
        <Text style={[styles.genderText, selected ? styles.selectedText : '']}>
          {buttonText}
        </Text>
      </View>
    </TouchableOpacity>
  );
};

const styles = StyleSheet.create({
  genderButton: {
    backgroundColor: '#fff',
    borderColor: '#f5f6fa',
    alignItems: 'center',
    justifyContent: 'center',
    width: 90,
    height: 30,
    borderRadius: 15,
    borderWidth: 1,
  },
  selectedButton: {
    backgroundColor: '#e9ecf5',
  },
  genderText: {
    backgroundColor: 'transparent',
    fontSize: 12,
    fontWeight: '700',
    color: '#a5b0c0',
  },
  selectedText: {
    color: '#5c646e',
  },
});

export default GenderButton;
