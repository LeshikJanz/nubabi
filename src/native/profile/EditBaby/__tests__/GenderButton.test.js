import 'react-native';
import React from 'react';
import GenderButton from '../GenderButton';
import renderer from 'react-test-renderer';

test('it renders correctly when selected', () => {
  const tree = renderer
    .create(<GenderButton selected buttonText="GIRL" />)
    .toJSON();

  expect(tree).toMatchSnapshot();
});

test('it renders correctly when not selected', () => {
  const tree = renderer
    .create(<GenderButton selected={false} buttonText="BOY" />)
    .toJSON();

  expect(tree).toMatchSnapshot();
});
