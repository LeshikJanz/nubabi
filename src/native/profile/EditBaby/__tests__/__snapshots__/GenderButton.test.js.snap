// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it renders correctly when not selected 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  hitSlop={undefined}
  isTVSelectable={true}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "#fff",
          "borderColor": "#f5f6fa",
          "borderRadius": 15,
          "borderWidth": 1,
          "height": 30,
          "justifyContent": "center",
          "width": 90,
        },
        Object {
          "marginRight": 20,
        },
        "",
      ]
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
            "color": "#a5b0c0",
            "fontSize": 12,
            "fontWeight": "700",
          },
          "",
        ]
      }
    >
      BOY
    </Text>
  </View>
</View>
`;

exports[`it renders correctly when selected 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  hitSlop={undefined}
  isTVSelectable={true}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "#fff",
          "borderColor": "#f5f6fa",
          "borderRadius": 15,
          "borderWidth": 1,
          "height": 30,
          "justifyContent": "center",
          "width": 90,
        },
        Object {
          "marginRight": 20,
        },
        Object {
          "backgroundColor": "#e9ecf5",
        },
      ]
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
            "color": "#a5b0c0",
            "fontSize": 12,
            "fontWeight": "700",
          },
          Object {
            "color": "#5c646e",
          },
        ]
      }
    >
      GIRL
    </Text>
  </View>
</View>
`;
